editor/src/core/model/project-file-utils.ts

```diff
@@ -644,44 +639,44 @@ export function switchToFileType(from: ProjectFile, to: ProjectFileType): Projec
 }
 
 export function uniqueProjectContentID(
-  startingID: string,
+  filename: string,
   projectContents: ProjectContentTreeRoot,
 ): string {
-  const startingIDCorrected = correctProjectContentsPath(startingID)
-  if (getContentsTreeFileFromString(projectContents, startingIDCorrected) != null) {
-    const firstIndexOfFullStop = startingIDCorrected.indexOf('.')
-    if (firstIndexOfFullStop === -1) {
-      let counter = 2
-      // eslint-disable-next-line no-constant-condition
-      while (true) {
-        const possibleNewID = `${startingIDCorrected}_${counter}`
-        if (getContentsTreeFileFromString(projectContents, possibleNewID) != null) {
-          counter += 1
-        } else {
-          return correctProjectContentsPath(possibleNewID)
-        }
-      }
-    } else {
-      // Kinda assume it's a filename.
-      const [prefix, suffixWithFullStop] = splitAt(firstIndexOfFullStop, startingIDCorrected)
-      const suffix = suffixWithFullStop.slice(1)
-      let counter = 2
-      // eslint-disable-next-line no-constant-condition
-      while (true) {
-        const possibleNewID = `${prefix}_${counter}.${suffix}`
-        if (getContentsTreeFileFromString(projectContents, possibleNewID) != null) {
-          counter += 1
-        } else {
-          return correctProjectContentsPath(possibleNewID)
-        }
+  const startingIDCorrected = correctProjectContentsPath(filename)
+  const fileWithSameNameExistsAlready =
+    getContentsTreeFileFromString(projectContents, startingIDCorrected) != null
+
+  if (!fileWithSameNameExistsAlready) {
+    return startingIDCorrected
+  }
+
+  const parts = getFilenameParts(startingIDCorrected)
+  if (parts == null) {
```

Comment:
It looks like both paths of this `if` statement are doing almost exactly the same thing, with the only difference being the value of `possibleNewID`, in which case you could just set that via a ternary

Reply:
or, I could generalise `getFilenameParts` so that it doesn't fail if a file doesn't have an extension, in which case the two branches will do the exact same work (I wanted to do this anyway but decided against because it felt like scope creep), would work instead?